worlds/tunic/er_scripts.py

```diff
@@ -124,12 +129,19 @@ def pair_portals(world: "TunicWorld") -> Dict[Portal, Portal]:
     two_plus: List[Portal] = []
     logic_rules = world.options.logic_rules.value
     player_name = world.multiworld.get_player_name(world.player)
+    portal_map = portal_mapping.copy()
 
     shop_scenes: Set[str] = set()
     shop_count = 6
-    if world.options.fixed_shop.value:
-        shop_count = 1
+    if world.options.fixed_shop:
+        shop_count = 0
         shop_scenes.add("Overworld Redux")
+    else:
+        # if fixed shop is off, remove this portal
+        for portal in portal_map:
+            if portal.name == "Ziggurat Lower Falling Entrance":
+                portal_map.remove(portal)
+                break
```

Comment:
Here you can't do that fun trick from above (unless you wanna refactor the code a bit), however I would still recommend the comprehension bc it's shorter
This is faster though (I timed it with timeit), so if you want it to be super fast, you can keep this